apps/prairielearn/elements/pl-string-input/pl-string-input.mustache

```diff
@@ -112,9 +157,41 @@
 </span>
 {{/error}}
 {{^error}}
-{{#label}}<span>{{{label}}}</span>{{/label}}
-<code class="user-output">{{a_sub}}</code>
-{{#suffix}}<span>{{suffix}}</span>{{/suffix}}
+{{^multiline}}
+    {{#label}}<span>{{{label}}}</span>{{/label}}
+    <code class="user-output">{{a_sub}}</code>
+    {{#suffix}}<span>{{suffix}}</span>{{/suffix}}
+{{/multiline}}
+{{#multiline}}
+    {{#label}}
+    <label class="form-label" for="pl-string-input-submission-{{uuid}}">{{{label}}}</label>
+    {{/label}}
+    <span class="input-group">
+        <textarea
+            wrap="soft"
+            type="text"
+            inputmode="text"
+            class="form-control multiline-input"
+            style="resize:none;"
+            disabled
+            id="pl-string-input-submission-{{uuid}}"
+        >{{a_sub}}</textarea>
+    </span>
+    {{#suffix}}
+    <label class="form-label" for="pl-string-input-submission-{{uuid}}">{{{suffix}}}</label>
+    {{/suffix}}
+{{/multiline}}
+
+<!-- Show submitted answer submission was parsed from -->
+<a href="javascript:void(0);" role="button" class="ml-1 btn btn-sm btn-secondary small border"
+    data-placement="auto" data-trigger="focus" data-toggle="popover" data-html="true"
+    title="Raw String" tabindex="0"
+    data-content="Invisible unicode characters are represented as
+    <code>&lt; U+xxxx &gt;</code>:<br>
+    <pre>{{escaped_submitted_answer}}</pre>">
+    <i class="fa fa-question-circle" aria-hidden="true"></i>
+</a>
```

Comment:
Overall, I think this PR is almost ready to go! The only things left are messing with formatting on a few things, which I would greatly appreciate a small amount of guidance with.

- Right now, I'm not sure whether the display of the suffix looks weird or not, the spacing is slightly different than the top label, but I'm not sure if this is standard.
- Is there a way to avoid having to put spaces in the first code tag in this popover? When I remove the spaces, the `<` and `>` don't display for some reason:

![image](https://github.com/PrairieLearn/PrairieLearn/assets/1345068/28ce188a-2205-47b8-864b-c3cc67f07645)

Other than that I think the behavior here is pretty reasonable and overall makes this element more straightforward.

Reply:
> Is there a way to avoid having to put spaces in the first code tag in this popover? When I remove the spaces, the < and > don't display for some reason:

Are you escaping the lt/gt? Assuming this is inside an argument, you may need to escape twice (`&amp;lt;` for `<`).

Reply:
The code for this popover is in the lines just above. I didn't need to escape twice to get it to work, but it stops working once I get rid of the surrounding whitespace.

Reply:
You do in fact have to escape this for this to work correctly (it's a happy accident that things work without escaping). It'd have to look something like `&lt;code&gt;&amp;lt;U+xxxx&amp;gt;&lt;/code&gt;`. Note how `<U` is first escaped to `&lt;U` and then again to `&amp;lt;U`. It's gross, but it works.

Reply:
Fix added here: https://github.com/PrairieLearn/PrairieLearn/pull/9438/commits/7cf4747c7a558d3012f7c664b1bd483e6bb00915